src/GraphQL.Conventions/Execution/UserContextExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using GraphQL.Execution;
+
+namespace GraphQL.Conventions.Execution
+{
+    public static class UserContextExtensions
+    {
+        public static IUserContext GetUserContext(this IProvideUserContext context) => GetValue<IUserContext>(context.UserContext);
+        public static void SetUserContext(this IProvideUserContext context, IUserContext value) => SetValue(context.UserContext, value);
+
+        public static IDependencyInjector GetDependencyInjector(this IProvideUserContext context) => GetValue<IDependencyInjector>(context.UserContext);
+        public static void SetDependencyInjector(this IProvideUserContext context, IDependencyInjector value) => SetValue(context.UserContext, value);
+
+        private static T GetValue<T>(IDictionary<string, object> dictionary)
+        {
+            var key = typeof(T).FullName;
+            return dictionary != null && dictionary.ContainsKey(key) 
+                ? (T) dictionary[key] 
+                : default;
```

Comment:
`TryGetValue()` ?

Reply:
Because TryGetValue use an out param, it doesn't really make the code shorter or more readable.

Reply:
I makes code to perform only one dictionary lookup instead of two.

Reply:
Agree, one lookup is better than two. You can still do it in one line of code, nearly the same as it’s written now